tests/test_stdiomgr_base.py

```diff
@@ -46,21 +46,54 @@ def test_capture_stdout(convert_newlines):
 
 
 def test_capture_stderr(convert_newlines, skip_warnings):
-    """Confirm stderr capture."""
+    """Confirm warnings under catch_warnings appear in stderr."""
     if skip_warnings:
         pytest.skip("Skip warning tests")
 
     with stdio_mgr() as (i, o, e):
         w = "This is a warning"
 
         with warnings.catch_warnings():
+            # catch_warnings should not have replaced the show warning impl
+            # with an instance method of list.append, but pytest may have.
+            try:
+                impl = warnings._showwarnmsg_impl
+            except AttributeError:
+                # Python < 3.6
+                impl = warnings.showwarning
+            assert impl.__name__ != "append", "Use pytest -p no:warnings"
```

Comment:
Looks like the goal of this try/except/assert is to provide an explicit diagnostic for the situation where someone attempts a pytest invocation without one or the other of `--skipwarnings` or `-p no:warnings`.

If so, might this be better implemented as an `autouse=True`, `scope="session"` fixture in `conftest.py`?